streaming/storage/subscription.go

```diff
@@ -67,3 +69,49 @@ func (sub *Subscription) Close() error {
 		return err
 	}
 }
+
+func ConcatSubscriptions(ctx context.Context, subs ...*Subscription) *Subscription {
+	count := len(subs)
+
+	channels := make([]reflect.SelectCase, len(subs)*2+1)
+	for i := range subs {
+		channels[i] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(subs[i].changes)}
+	}
+	for i := range subs {
+		channels[count+i] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(subs[i].errors)}
+	}
+	return NewSubscription(ctx, func(ctx context.Context, changes chan<- struct{}) error {
+		channels[len(channels)-1] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(ctx.Done())}
+		var closedErr error
+	loop:
```

Comment:
wait what? isn't there any other way to handle this? :P

Reply:
No, and it's a standard pattern.
```
loop:
for {
  select {
  // ...
  break loop
  }
}
```
cause otherwise you'd be breaking out of the channel.

Reply:
alrighty